server/accounts.go

```diff
@@ -3153,7 +3156,12 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 			c.mu.Lock()
 			sk := c.user.SigningKey
 			c.mu.Unlock()
-			if sk != "" && !a.hasIssuer(sk) {
+			if sk == _EMPTY_ {
+				continue
+			}
+			if _, ok := alteredScope[sk]; ok {
+				c.closeConnection(AuthenticationViolation)
```

Comment:
so here we are simply d/c because their configuration changed, and a reconnect will make them have the correct permissions?

Reply:
this seems reasonable.